sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -122,6 +124,12 @@ class SparkSession private(
       .getOrElse(SQLConf.getFallbackConf)
   })
 
+  /**
+   * A map to hold the mapping from user-defined tags to the real tags attached to Jobs.
+   * Real tag have the current session ID attached: `"tag1" -> s"spark-$sessionUUID-tag1"`.
+   */
+  private val userDefinedToRealTagsMap: ConcurrentHashMap[String, String] = new ConcurrentHashMap()
```

Comment:
Shouldn't we inherit this when somebody clone a session from a parent session?

Reply:
Otherwise, jobs like broadcast hash joins won't be cancelled IIRC.

Reply:
Good point. I'll change the code.
Btw, how should we handle session IDs? Should calling `interruptAll` from parent sessions terminate jobs from child sessions?

Reply:
Tags set in parent sessions should be set in children sessions too. Can we match this with Spark Connect cancellation implementation? It's already doing so.

Reply:
Done!